### PR TITLE
Allow unsigned int array types for arrays with dtype=int

### PR DESCRIPTION
**Pull request by fwkoch**

### CODE REVIEW

File: properties/math.py [hunk 1]
```diff
@@ -132,7 +132,8 @@ def validate(self, instance, value):
                 'Array validation is only implmented for wrappers that are '
                 'subclasses of numpy.ndarray'
             )
-        if value.dtype.kind not in (TYPE_MAPPINGS[typ] for typ in self.dtype):
+        allowed_kinds = ''.join(TYPE_MAPPINGS[typ] for typ in self.dtype)
```
Comment (bsmithyman) [inline properties/math.py]: Okay, so this should be safe b/c the setter on `.dtype` pre-validates that the specified dtypes are in the `TYPE_MAPPINGS` anyway, right?

File: properties/math.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 from .utils import ValidationError
 
 TYPE_MAPPINGS = {
-    int: 'i',
+    int: 'ui',
```
Comment (bsmithyman) [inline properties/math.py]: I guess I don't quite follow here... why doesn't this break signed `int`s?

Comment (fwkoch) [reply properties/math.py]: Unsigned ints have `kind='u'`, signed have `kind='i'` - so the check ends up as `kind in 'ui'` or if you do something like `..., dtype=(int, float, bool)` the check will end up as `kind in 'uifb'`. Make sense?

Comment (bsmithyman) [reply properties/math.py]: Yep, thanks :)

**Pull request merged**